suse_migration_services/units/kernel_reboot.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2018 SUSE Linux LLC.  All rights reserved.
+#
+# This file is part of suse-migration-services.
+#
+# suse-migration-services is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# suse-migration-services is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with suse-migration-services. If not, see <http://www.gnu.org/licenses/>
+#
+
+# project
+import re
+import os
+from suse_migration_services.command import Command
+from suse_migration_services.defaults import Defaults
+
+from suse_migration_services.exceptions import (
+    DistMigrationKernelRebootException
+)
+
+
+def main():
+    """
+    DistMigration reboot with  new kernel
+
+    Reads the info of the new kernel and loads the latest (new migrated kernel)
+    version for rebooting.
+    """
+    target_kernel = Defaults.get_target_kernel()
+    target_initrd = Defaults.get_target_initrd()
+    try:
+        Command.run(
+            ['kexec', '--unload']
+        )
+        Command.run(
+            [
+                'kexec',
+                '--load', target_kernel,
+                '--initrd', target_initrd,
+                '--command-line', _get_cmdline(target_kernel)
+            ]
+        )
+        Command.run(
+            ['kexec', '--exec']
+        )
+    except Exception as issue:
+        Command.run(
+            ['kexec', '--unload']
+        )
+        raise DistMigrationKernelRebootException(
+            'Kernel rebooting failed with {0}'.format(
+                issue
+            )
+        )
+
+
+def _get_cmdline(vmlinuz):
+    grub_config_file_path = Defaults.get_grub_config_file()
+    if not os.path.exists(grub_config_file_path):
+        raise DistMigrationKernelRebootException(
+            'Could not find {0} to load the kernel options'.format(
+                grub_config_file_path
+            )
+        )
+    pattern = r'(?<=linux\t).*'
```

Comment:
just asking but does ```\t``` identify a tab ? if so I would not rely on that format, it should at least match on a char class with tabs and spaces

Reply:
grub config does not enforce tabs like e.g make

Reply:
Yes, it does identify tab.

Reply:
I will refactor the method with the parameter to make it more generic.